### PR TITLE
TF-TRT Add Erf unary op converter

### PR DESCRIPTION
**Pull request by tfeher**
Add conversion for Erf op.

Tracker: #45481 
Tagging @bixia1 for review and @DEKHTIARJonathan

### CODE REVIEW

**Review by bixia1 — APPROVED** [review]

**Review by bixia1 — CHANGES_REQUESTED** [review]
base_test.py fails. convert_nodes_test also fails.

SimpleMultiEnginesTest.testTfTrtV2_OfflineConversion_DynamicEngine_FP16_NoCalibration
1.800s
Traceback (most recent call last):
  File "/build/work/b3f5bf78ffa7769742a817f7ec91baa32dba/google3/runfiles/google3/third_party/tensorflow/python/framework/test_util.py", line 1443, in decorated
    return f(self, *args, **kwargs)
  File "/build/work/b3f5bf78ffa7769742a817f7ec91baa32dba/google3/runfiles/google3/third_party/tensorflow/python/compiler/tensorrt/test/tf_trt_integration_test_base.py", line 1108, in _Test
    self.RunTest(run_params)
  File "/build/work/b3f5bf78ffa7769742a817f7ec91baa32dba/google3/runfiles/google3/third_party/tensorflow/python/compiler/tensorrt/test/tf_trt_integration_test_base.py", line 1044, in RunTest
    rtol=self.ExpectedRelativeTolerance(run_params))
  File "/build/work/b3f5bf78ffa7769742a817f7ec91baa32dba/google3/runfiles/google3/third_party/tensorflow/python/framework/test_util.py", line 1246, in decorated
    return f(*args, **kwds)
  File "/build/work/b3f5bf78ffa7769742a817f7ec91baa32dba/google3/runfiles/google3/third_party/tensorflow/python/framework/test_util.py", line 2733, in assertAllClose
    self._assertAllCloseRecursive(a, b, rtol=rtol, atol=atol, msg=msg)
  File "/build/work/b3f5bf78ffa7769742a817f7ec91baa32dba/google3/runfiles/google3/third_party/tensorflow/python/framework/test_util.py", line 2673, in _assertAllCloseRecursive
    (path_str, path_str, msg))
  File "/build/work/b3f5bf78ffa7769742a817f7ec91baa32dba/google3/runfiles/google3/third_party/tensorflow/python/framework/test_util.py", line 2628, in _assertArrayLikeAllClose
    a, b, rtol=rtol, atol=atol, err_msg="\n".join(msgs), equal_nan=True)
  File "/build/work/b3f5bf78ffa7769742a817f7ec91baa32dba/google3/runfiles/google3/third_party/py/numpy/testing/_private/utils.py", line 1501, in assert_allclose
    verbose=verbose, header=header, equal_nan=equal_nan)
  File "/build/work/b3f5bf78ffa7769742a817f7ec91baa32dba/google3/runfiles/google3/third_party/py/numpy/testing/_private/utils.py", line 757, in assert_array_compare
    flagged = func_assert_same_pos(x, y, func=isnan, hasval='nan')
  File "/build/work/b3f5bf78ffa7769742a817f7ec91baa32dba/google3/runfiles/google3/third_party/py/numpy/testing/_private/utils.py", line 733, in func_assert_same_pos
    raise AssertionError(msg)
AssertionError: 
Not equal to tolerance rtol=0.01, atol=0.01
Mismatched value: a is different from b. 
not close where = (array([0, 0, 0, ..., 0, 0, 0]), array([0, 0, 0, ..., 0, 0, 0]), array([ 0,  0,  0, ..., 99, 99, 99]), array([ 0,  0,  0, ..., 11, 11, 11]), array([ 0,  1,  2, ...,  1,  4, 10]), array([2, 2, 0, ..., 3, 3, 3]))
not close lhs = [nan nan nan ... nan nan nan]
not close rhs = [-inf -inf -inf ... -inf -inf -inf]
not close dif = [nan nan nan ... nan nan nan]
not close tol = [inf inf inf ... inf inf inf]
dtype = float32, shape = (1, 1, 100, 12, 12, 6)
x and y nan location mismatch:
 x: array([[[[[[-3.820660e+00, -2.609254e+00,           nan,
            -6.645927e+00, -2.989990e+00, -2.343089e+00],
           [-4.728037e+00,  3.076724e+00,           nan,...
 y: array([[[[[[-3.820661e+00, -2.609254e+00,          -inf,
            -6.645927e+00, -2.989990e+00, -2.343089e+00],
           [-4.728037e+00,  3.076723e+00,          -inf,...

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc [hunk 1]
```diff
@@ -5990,6 +5990,7 @@ TEST_P(OpConverter_FP32_Test, ConvertUnary) {
   ADD_OP("Cos", ops::Cos, std::cos);
   ADD_OP("Cosh", ops::Cosh, std::cosh);
   ADD_OP("Exp", ops::Exp, std::exp);
+  ADD_OP("Erf", ops::Erf, std::erf);
```
Comment (bixia1) [inline tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc:5993]: Need to protect this.
Also need to fix these two lines:
https://github.com/tensorflow/tensorflow/blob/9e26f3b56acbff18f01da8afce6b571739f59ef3/tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc#L684
https://github.com/tensorflow/tensorflow/blob/9e26f3b56acbff18f01da8afce6b571739f59ef3/tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc#L719

**Review by tfeher — COMMENTED** [review]
Thanks @bixia1 for pointing out the problem with convert_nodes_test, I have fixed it. I could not reproduce the python problem. What GPU was used for the failing test?

**Comment by bixia1** [comment]
I am able to fix the test by replacing erf with 1-erfc, instead of replacing erf with erfc.

**Review by bixia1 — APPROVED** [review]

**Pull request merged**